### PR TITLE
Telemetry Perf Tests added

### PR DESCRIPTION
**Pull request by msk2208**
## Description
- Added Telemetry Perf 10k tests for Mstest,Xunit and Nunit
- Included a TelemetryPerfTestBase class 
- Post the telemetry details to AppInsights.

Sample data being posted

Dimensions data(Properties)
```
VS.TestDiscovery.ParallelEnabled False
VS.TestDiscovery.DiscoveryState Completed
VS.TestRun.TestPlatformVersion 15.7.0-preview-20180315-02
VS.TestRun.TargetFramework .NETFramework,Version=v4.5
VS.TestRun.TargetPlatform X86
VS.TestRun.TargetDevice Local Machine
VS.TestRun.TargetOS Microsoft Windows NT 10.0.16299.0
```

Measurement data(Metrics)
```
VS.TestDiscovery.TimeTakenAdapter.executor://cppunittestexecutor/v1 0.0468793
VS.TestDiscovery.TimeTakenAdapter.executor://codedwebtestadapter/v1 0.1406143
VS.TestDiscovery.TimeTakenAdapter.executor://mstestadapter/v1 0.0156247
VS.TestDiscovery.TimeTakenToLoadAdaptersInSec 0
VS.TestDiscovery.TimeTakenInSecByAllAdapters 2.7500567
VS.TestDiscovery.AdaptersDiscoveredCount 4
VS.TestDiscovery.TotalTimeTakenInSec 3.5469674
VS.TestDiscovery.AdaptersUsedCount 1
VS.TestDiscovery.NumberOfSources 1
VS.TestDiscovery.TotalTests 10,000
VS.TestRun.MaxCPUcount 1
VS.TestDiscovery.TotalTestsDiscovered.executor://mstestadapter/v2 10,000
VS.TestDiscovery.TimeTakenAdapter.executor://mstestadapter/v2 2.5469384
```

### CODE REVIEW

**Comment by msk2208** [comment]
Please verify and approve it.

**Comment by karanjitsingh** [comment]
@dotnet-bot test Windows_NT / Release Build

File: test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            client = new TelemetryClient();
+            TelemetryConfiguration.Active.InstrumentationKey = TelemetryInstrumentationKey;
+        }
+
+        /// <summary>
+        /// Used for posting the telemtery to AppInsights
+        /// </summary>
+        /// <param name="testCaseName"></param>
+        /// <param name="handlerMetrics"></param>
+        public void PostTelemetry(string testCaseName, IDictionary<string,object> handlerMetrics)
```
Comment (smadala) [inline test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs]: Rename testcaseName to perfscenario.

File: test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                if (double.TryParse(stringValue, out var doubleValue))
+                {
+                    metrics.Add(entry.Key, doubleValue);
+                }
+                else
+                {
+                    properties.Add(entry.Key, stringValue);
+                }
+            }
+            this.client.TrackEvent(testCaseName, properties, metrics);
```
Comment (smadala) [inline test/Microsoft.TestPlatform.PerformanceTests/TranslationLayer/TelemetryPerfTestBase.cs]: Add sample data point to PR description.

**Review by smadala — APPROVED** [review]

**Comment by singhsarab** [comment]
@dotnet-bot  test Windows_NT / Debug Build

**Comment by singhsarab** [comment]
@dotnet-bot test Windows_NT / Debug Build

**Comment by msk2208** [comment]
Closing this, covered with https://github.com/Microsoft/vstest/pull/1493

**Pull request closed without merging**